syntax = "proto3";

service ParquetStream {
    rpc ProcessFiles (FileRequest) returns (stream FileResponse);
}

message FileRequest {
    repeated string file_paths = 1;
}

message FileResponse {
    string file_path = 1;
		int32 total_row_groups = 2;
    int32 row_group_index = 3;
    string data = 4;
}
